Remove merchant email notes

psealock opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/woocommerce/woocommerce/issues/44836

In https://github.com/woocommerce/woocommerce-admin/pull/5922 Merchant emails were added to notify merchants about various things relating to their site. The effort was abandoned and this PR removes the note type as well as the email functionality.

Note that the `Note::E_WC_ADMIN_NOTE_EMAIL` type has already been deprecated, but I'm leaving the type in case any plugins make use of it still. I also removed it from `get_allowed_types` so that any new notes may not have the `email` type.

Also note that the removed functionality is different than merchant emails that fire when things like an order has been placed. @NeosinneR could you confirm this?

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Although its somewhat hard to test the lack of this functionality, I think the main thing here is to test if any errors occur where an install has an unactioned `email` type notification lingering in their database.

1. Create a JN or local instance
2. See what notes you have in the database

```mysql
SELECT note_id, type, is_read, is_deleted FROM wp_wc_admin_notes LIMIT 20;
```

3. Find one that is not read, nor deleted and change its type to `email`

```mysql
UPDATE wp_wc_admin_notes SET type = 'email' WHERE note_id = 1;
```
4. Run the daily cron, WCA Test Helper > Tools > Run wc_admin_daily job and click run.
5. Ensure no errors occur.
6. Visit WooCommerce > Settings > Emails and ensure no `Store management insights` section exists.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

psealock committed:
remove notes

psealock committed:
remove tests and references

psealock committed:
Merge branch 'trunk' into try/remove-merchant-email-notes

psealock committed:
changelog

psealock committed:
remove email notes test helper

psealock committed:
Update try-remove-merchant-email-notes

psealock committed:
update tests

psealock committed:
Merge branch 'try/remove-merchant-email-notes' of github.com:woocommerce/woocommerce into try/remove-merchant-email-notes

psealock committed:
Merge branch 'trunk' into try/remove-merchant-email-notes

psealock committed:
update tests

psealock committed:
Merge branch 'try/remove-merchant-email-notes' of github.com:woocommerce/woocommerce into try/remove-merchant-email-notes

moon0326 reviewed (approved):
LGTM and tested well 👍 🚀

chihsuan reviewed (commented):
> Find one that is not read, nor deleted and change its type to email
> UPDATE wp_wc_admin_notes SET type = 'email' WHERE note_id = 1;

@psealock I changed the note type to email for other notes in db and got the following error. It looks like this only happens for notes where the status is `pending`. Is this expected?

```
[27-Feb-2025 04:20:30 UTC] Exception caught in Automattic\WooCommerce\Admin\Notes\Notes::get_note. The admin note type prop (email) is not one of the supported types.. Args: Array
(
    [0] => 2
)
..
[27-Feb-2025 04:20:30 UTC] Exception caught in Automattic\WooCommerce\Admin\Notes\Notes::get_note. The admin note type prop (email) is not one of the supported types.. Args: Array
(
    [0] => 3
)
```

```
+---------+-------+---------------------------------------+------------+
| note_id | type  | name                                  | status     |
+---------+-------+---------------------------------------+------------+
|       1 | email | wc-refund-returns-page                | unactioned |
|       2 | email | wayflyer_bnpl_q4_2021                 | pending    |
|       3 | email | wc_shipping_mobile_app_usps_q4_2021   | pending    |
```

NeosinneR reviewed (commented):
@psealock 

>Also note that the removed functionality is different than merchant emails that fire when things like an order has been placed. 

That's correct. Order emails and similar are not affected by this. 

I noticed, that there is still [`EmailNotification`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/Notes/EmailNotification.php) with associated templates ([`html-merchant-notification.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/react-admin/emails/html-merchant-notification.php) and [`plain-merchant-notification.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/react-admin/emails/plain-merchant-notification.php)) present, which are not used anywhere anymore - they were only used in now removed `MerchantEmailNofications` class (introduced [in the same PR](https://github.com/woocommerce/woocommerce-admin/pull/5922/files) when merchant notifications were added). It being in internal namespace, do you think we can also remove them?

psealock committed:
Merge branch 'trunk' into try/remove-merchant-email-notes

psealock committed:
fix merge mistake

psealock committed:
remove deprecated notes

psealock commented:
> @psealock I changed the note type to email for other notes in db and got the following error. It looks like this only happens for notes where the status is pending. Is this expected?

@chihsuan no, this isn't expected - thanks for picking it up. I'm not entirely too sure how to handle this, but in [remove deprecated notes](https://github.com/woocommerce/woocommerce/pull/55660/commits/a71beabca71066517d84afec4a0c4684d49c2b0c) I added a `get_deprecated_types` method where we can handle deprecated notes specifically. I like this idea, but wasn't too sure how to deal with it. I decided upon just deleting the note. Is this too heavy handed?

### Testing

1. Set some notes to `email` type.

```mysql
UPDATE wp_wc_admin_notes SET type = 'email' WHERE note_id = 3;
```
2. Make sure the status is `pending`
3. Run the daily cron, WCA Test Helper > Tools > Run wc_admin_daily job and click run.
4. See the error logged.
5. Repeat step 3, there should not be another error logged.
6. Confirm your note is no longer in the database
```mysql
SELECT note_id, type, is_read, is_deleted, status FROM wp_wc_admin_notes LIMIT 20;
```

psealock committed:
remove unused email template files

psealock commented:
>I noticed, that there is still [EmailNotification](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/Notes/EmailNotification.php) with associated templates ([html-merchant-notification.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/react-admin/emails/html-merchant-notification.php) and [plain-merchant-notification.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/react-admin/emails/plain-merchant-notification.php)) present, which are not used anywhere anymore - they were only used in now removed MerchantEmailNofications class (introduced [in the same PR](https://github.com/woocommerce/woocommerce-admin/pull/5922/files) when merchant notifications were added). It being in internal namespace, do you think we can also remove them?

Good find @NeosinneR ! Indeed I checked manually and had Cursor check for references in the codebase and found none. I removed those files [remove unused email template files](https://github.com/woocommerce/woocommerce/pull/55660/commits/ff540d81dc969ed0fafc27dbcac42d7074e98251) and I also think its safe to do so because we're operating under the `Internal` namespace.

adrianduffell commented:
> I decided upon just deleting the note. Is this too heavy handed?

I like the idea of deleting the notes, but think it could be too heavy-handed having such a side-effect in `set_type()`. Alternatively maybe existing email notes could be deleted with the DB update system?

chihsuan commented:
>  Alternatively maybe existing email notes could be deleted with the DB update system?

This sounds like a good idea. 👍

psealock committed:
don't delete notes on error

psealock committed:
run db update operation

psealock committed:
Merge branch 'trunk' into try/remove-merchant-email-notes

psealock commented:
>  Alternatively maybe existing email notes could be deleted with the DB update system?

Good idea. 

### Updated testing instructions

1. Mark a note as type email

```mysql
UPDATE wp_wc_admin_notes SET type = 'email' WHERE note_id = 3;
```
3. Update the verison, `wp option update woocommerce_db_version "9.8.0"`
4. Run the update, `wp wc update`
5. See that there are no longer email type notes

```mysql
SELECT note_id, type, is_read, is_deleted, status FROM wp_wc_admin_notes WHERE type = 'email';
```
6. Ensure no errors are logged.

chihsuan reviewed (approved):
> Updated testing instructions

Thanks @psealock tested well!

### Test environment

<detail>
WordPress: 6.7.2
Theme: Twenty Twenty-Five Version: 1.1
WooCommerce: Development version with try/remove-merchant-email-notes branch
PHP: 8.3
Web Server: Nginx
Browser: Brave
OS: macOS Sequoia 15.3.1
</detail>

### Scenarios tested
| Scenario | Result | Screenshot |
|----------|---------|------------|
| Existing email type notes handling | ✅ Notes properly removed via DB update | N/A |
| WooCommerce Settings > Emails section | ✅ 'Store management insights' section removed | N/A |
| Daily cron job with email type notes | ✅ No errors after cleanup | N/A |
